api/renderers.py

```diff
@@ -46,11 +50,201 @@ def extract_attributes(fields, resource):
+                    )
+                    relation_data.update(
+                        {
+                            'meta': {
+                                'count': len(resource.get(field_name))
+                            }
+                        }
+                    )
+                    data.update({field_name: relation_data})
+                    continue
+
+                relation_data = list()
+                for nested_resource_instance in relation_instance:
+                    nested_resource_instance_type = (
+                        relation_type or
+                        utils.get_resource_type_from_instance(nested_resource_instance)
+                    )
+
+                    relation_data.append(OrderedDict([
+                        ('type', nested_resource_instance_type),
+                        ('id', encoding.force_text(nested_resource_instance.pk))
+                    ]))
+                data.update({
+                    field_name: {
+                        'data': cls.encode_ids(relation_data),
+                        'meta': {
+                            'count': len(relation_data)
+                        }
+                    }
+                })
+                continue
+
+            if isinstance(field, ListSerializer):
+                resolved, relation_instance = utils.get_relation_instance(resource_instance, source, field.parent)
+                if not resolved:
+                    continue
+
+                relation_data = list()
+
+                serializer_data = resource.get(field_name)
+                resource_instance_queryset = list(relation_instance)
+                if isinstance(serializer_data, list):
+                    for position in range(len(serializer_data)):
+                        nested_resource_instance = resource_instance_queryset[position]
+                        nested_resource_instance_type = (
+                            relation_type or
+                            utils.get_resource_type_from_instance(nested_resource_instance)
+                        )
+
+                        relation_data.append(OrderedDict([
+                            ('type', nested_resource_instance_type),
+                            ('id', encoding.force_text(nested_resource_instance.pk))
+                        ]))
+
+                    data.update({field_name: {'data': cls.encode_ids(relation_data)}})
+                    continue
+
+            if isinstance(field, Serializer):
+                resolved, relation_instance = utils.get_relation_instance(resource_instance, source, field.parent)
+                if not resolved:
+                    continue
+
+                data.update({
+                    field_name: {
+                        'data': (
+                            OrderedDict([
+                                ('type', relation_type),
+                                ('id', cls.encode_id(resource_instance.pk, relation_type))
```

Comment:
Here I could use `encoding.force_text(IDObfuscator.encode(resource_instance))` instead of `cls.encode_id(resource_instance.pk, relation_type)` but it seems more clear to have all of the encoding use the same method. That way if `IDObfuscator` ever changes, we will only have to change the code in one place.